### PR TITLE
Modify autocast doc according #iotagent-node-lib/1324

### PR DESCRIPTION
**Pull request by mapedraza**
This PR modifies the doc in order to align it after changes in https://github.com/telefonicaid/iotagent-node-lib/pull/1324

### CODE REVIEW

File: docs/usermanual.md [hunk 1]
```diff
@@ -146,20 +146,15 @@ t|10|s|true|l|78.8
...
 
-In addition, the device has to be provisioned using the right types for the attributes to be cast, which are:
-
--   Type "Number" for integer or float numbers
--   Type "Boolean" for boolean
--   Type "None" for null
-
-As a consequence of the above, note the casting to JSON native format doesn't work for autoprovisioned devices as
-autoprovisioning doesn't allow to provide explicit types for each attribute (all them are considered of default type
-"string").
```
Comment (mapedraza) [inline docs/usermanual.md:162]: This need to be tested

Comment (fgalan) [reply docs/usermanual.md:162]: If we don't want to block this PR until de refactor in IOTA Lib gest completed, I'd suggest to move the comment to the .md itself, in the following way;

> **FIXME:** this need to be tested, once IOTA Lib 3.0.0 gets released and IOTA UL 2.0.0 (using it) gets released.

or something similar.

File: docs/usermanual.md [hunk 2]
```diff
@@ -136,6 +136,8 @@ the following meanings:
 
 #### Casting to JSON native format
 
+FIXME: this need to be tested, once IOTA Lib 3.0.0 gets released and IOTA UL 2.0.0 (using it) gets released.
```
Comment (fgalan) [inline docs/usermanual.md:139]:
```suggestion
**FIXME**: this need to be tested, once IOTA Lib 3.0.0 gets released and IOTA UL 2.0.0 (using it) gets released.
```

**Review by fgalan — APPROVED** [review]
LGTM

File: docs/usermanual.md [hunk 3]
```diff
@@ -146,20 +148,62 @@ t|10|s|true|l|78.8
...
 [configuration section of iotagent library](https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/installationguide.md#global-configuration)
 for further information.
 
-In addition, the device has to be provisioned using the right types for the attributes to be cast, which are:
+This functionality does not change the attribute type, using the type specified in the config group or device provision.
+    
+As an example, for a given measure:
+    
+```
+a|1|b|1.01|c|true|d|null|e|[1,2,3]|f|['a','b','c']g|{a:1,b:2,c:3}
```
Comment (mrutid) [inline docs/usermanual.md:164]: a|1|b|1.01|c|true|d|null|e|[1,2,3]|f|['a','b','c']g|{a:1,b:2,c:3}|h|I'm a string

File: docs/usermanual.md [hunk 4]
```diff
@@ -146,20 +148,62 @@ t|10|s|true|l|78.8
...
+    "value": [1,2,3]
+  },
+  "f": {
+    "type": "provisionedType",
+    "value": ["a","b","c"]
+  },
+  "g": {
+    "type": "provisionedType",
+    "value": {"a":1,"b":2,"c":3}
+  }
```
Comment (mrutid) [inline docs/usermanual.md:200]: "h": {
    "type": "provisionedType",
    "value": "I'm a string"
  }

**Pull request merged**